src/app/models/user.ts

```diff
@@ -0,0 +1,43 @@
+import {UserData} from './user-data';
+
+export class User {
+  private id: number;
+  private mail: string;
+  private password: string;
+  private userData: UserData;
+  constructor(
+    mail = 'default',
+    password = 'default',
+    id = -1,
+    userData = new UserData(),
+  ) {
+    this.id = id;
+    this.mail = mail;
+    this.password = password;
+    this.userData = userData;
+  }
+  get Id(): number {
+    return this.id;
+  }
+  set Id(id: number) {
+    this.id = id;
+  }
```

Comment:
altro tip per velocizzare la creazione di classi:

https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties

```suggestion
  constructor(
    private _mail,
    private _password,
    private _id,
    private _userData,
  ) { }
  
  get id() {return this._id;}
  // niente setter per l'id
```

Reply:
Va bene però non possiamo usare i campi dati in questa forma `_id` perchè non coinciderebbe con i campi dati nell'api